_includes/tutorials/message-ordering/kafka/markup/dev/make-application.adoc

```diff
@@ -0,0 +1,26 @@
+Before you create your full application code, let's highlight the most important `ProducerConfig` configuration parameters:
+
++++++
+<pre class="snippet"><code class="java">{% include_raw tutorials/message-ordering/kafka/code/tutorial-steps/dev/answer-code.java %}</code></pre>
++++++
+
+The first configuration sets the producer parameter https://kafka.apache.org/documentation/#enable.idempotence[enable.idempotence].
+When set to `true`, it enables an idempotent producer which ensures that exactly one copy of each message is written to the brokers, and in order.
+The default value is `enable.idempotence=false`, so you must explicitly set this to `enable.idempotence=true`.
+For further reading, please see https://cwiki.apache.org/confluence/display/KAFKA/KIP-98+-+Exactly+Once+Delivery+and+Transactional+Messaging#KIP98ExactlyOnceDeliveryandTransactionalMessaging-IdempotentProducerGuarantees[KIP-98].
+
+The second configuration sets the producer parameter https://kafka.apache.org/documentation/#acks[acks].
+The `KafkaProducer` uses the `acks` configuration to tell the leader broker how many acknowledgments to wait for to consider a produce request complete.
+You must set `acks=all` in order to use the idempotent producer, otherwise the producer cannot guarantee idempotence.
```

Comment:
`KafkaProducer` will automatically set `acks=all` if `enable.idempotence=true`:

```
[main] INFO org.apache.kafka.clients.producer.KafkaProducer - [Producer clientId=myApp] Overriding the default acks to all since idempotence is enabled.
```

Reply:
I was aware that setting `enable.idempotence=true` overrides some other producer settings. I was on the fence about whether to include them or not, and I decided to include them for a few reasons:

1. KT can _teach_ users about those settings
2. Sometimes being explicit is better practice because it prevents unintended behavior (user might miss the log message)...
3. Future proof: what if that behavior defaults change in Kafka, etc.

Maybe KT could say that `acks=all` is required for idempotency, so either the user must set it explicitly _or_ whatever setting is there will get overridden by the app.  WDYT?

Reply:
@mikebin revised copy available, WDYT?

Reply:
LGTM - thanks!